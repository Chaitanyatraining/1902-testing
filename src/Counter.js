import React, { useState } from 'react'

const Counter = () => {
    const [count, setCount] = useState(1)

  return (
    <div>
        <h2>Count</h2>
        <p data-testid='counter'>Count:{count}</p>
        <button data-testid='increment' onClick={()=>{setCount(count + 1)}}>Increment</button>
        <button data-testid='decrement' onClick={()=>{setCount(count - 1)}}>Decrement</button>
    </div>
  )
}

export default Counter