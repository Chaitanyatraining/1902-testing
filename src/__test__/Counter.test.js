import { render, screen, fireEvent } from '@testing-library/react';
import Counter from '../Counter';
import renderer from 'react-test-renderer'

describe('counter', ()=> {
    it('should increment counter', () => {
        render(<Counter />)
        const counter = screen.getByTestId('counter')
        const incrementBtn = screen.getByTestId('increment')
        const decrementBtn = screen.getByTestId('decrement')
    
        fireEvent.click(incrementBtn)
    
        expect(counter).toHaveTextContent('2')
    })
})

describe('counter snapshot', ()=>{
    it('should match DOM snapshot', ()=>{
        const tree = renderer.create(<Counter />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})



// test('increment', ()=>{
//     render(<Counter />)
//     const counter = screen.getByTestId('counter')
//     const incrementBtn = screen.getByTestId('increment')
//     const decrementBtn = screen.getByTestId('decrement')

//     fireEvent.click(incrementBtn)

//     expect(counter).toHaveTextContent('2')

//     // fireEvent.click(decrementBtn)
//     // expect(counter).toHaveTextContent('0')
// })