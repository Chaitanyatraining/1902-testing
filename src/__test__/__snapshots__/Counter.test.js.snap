// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`counter snapshot should match DOM snapshot 1`] = `
<div>
  <h2>
    Count
  </h2>
  <p
    data-testid="counter"
  >
    Count:
    1
  </p>
  <button
    data-testid="increment"
    onClick={[Function]}
  >
    Increment
  </button>
  <button
    data-testid="decrement"
    onClick={[Function]}
  >
    Decrement
  </button>
</div>
`;
